File: medium/sort_colors.go
package medium

func SortColors(nums []int) {
	freq := make(map[int]int)
	for _, num := range nums {
		freq[num]++
	}
	index := 0
	for i := range 3 {
		count := freq[i]
		for count > 0 {
			count--
			nums[index] = i
			index++
		}
	}
}

File: medium/sort_colors.py
from collections import Counter


def sort_colors(nums: list[int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    freq = Counter(nums)
    index = 0
    for i in range(3):
        count = freq.get(i, 0)
        while count > 0:
            nums[index] = i
            index += 1
            count -= 1
